pkg/kubelet/apis/config/types.go

```diff
@@ -568,3 +572,10 @@ type MemoryReservation struct {
 	NumaNode int32
 	Limits   v1.ResourceList
 }
+
+type MemorySwapConfiguration struct {
```

Comment:
I expected to see something in here indicating whether swap support was enabled, defaulting to the current behavior of the kubelet not allowing swap. Is this feature intended to unconditionally support swap once it graduates to beta/GA?

Reply:
In order to enable swap, one must also enable the `--fail-swap-on=false` flag on the kubelet. Otherwise, kubelet will not start when there is swap on the system.

So matrix is:

| |fail-swap-on=true|fail-swap-on=false|
|-|-|-|
|feature-flag=false|Kubelet won't start if swap is available|Kubelet will default to current behaviour|
|feature-flag=true|Kubelet won't start if swap is available|Kubelet will use swap config|

My plan is to eventually remove the feature flag, but there are no plans for removing the `--fail-swap-on` flag.

Reply:
ah, I missed the separate `failSwapOn` field in the kubelet config file. that's what I was looking for